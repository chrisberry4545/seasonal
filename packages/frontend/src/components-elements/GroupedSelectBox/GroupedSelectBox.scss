@import '../../styles/settings';
@import '../../styles/tools';

.c-grouped-select-box__group {
  margin-bottom: 2rem;
}

.c-grouped-select-box__group-name {
  display: block;
  margin-bottom: 0.75rem;
  padding-bottom: 0.5rem;
  border-bottom: 1px  solid $grey-med;
}

.c-grouped-select-box__option {
  display: block;
  margin-bottom: 0.25rem;
  opacity: 0.6;
}

  .c-grouped-select-box__option--selected {
    opacity: 1;
  }
