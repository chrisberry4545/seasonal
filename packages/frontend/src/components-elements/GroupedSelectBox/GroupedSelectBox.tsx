import React, { FC } from 'react';
import { IGroupedSelectOptions } from '@chrisb-dev/seasonal-shared-models';
import { BareButton, TextMedium, TextHeadingSmall } from '@chrisb-dev/seasonal-shared-frontend-components';
import './GroupedSelectBox.scss';

export const GroupedSelectBox: FC<{
  groups: IGroupedSelectOptions[] | undefined,
  onSelected?: (selectedValue: string) => void,
  'data-e2e'?: string
}> = ({
  groups,
  onSelected,
  ...rest
}) => (
  <div data-e2e={rest['data-e2e']}>
  {
    groups && groups.map((group) => (
      <div className='c-grouped-select-box__group' key={group.groupName}>
        <TextHeadingSmall className='c-grouped-select-box__group-name'>
          {group.groupName}
        </TextHeadingSmall>
        {
          group.selectOptions.map((option) =>
            <BareButton
              data-e2e={`select-option-${option.name}`}
              className={
                'c-grouped-select-box__option'
                + (option.isSelected
                  ? ' c-grouped-select-box__option--selected'
                  : ''
                )
              }
              key={option.value}
              onClick={() => onSelected && onSelected(option.value)}>
              <TextMedium>{option.name}</TextMedium>
            </BareButton>
          )
        }
      </div>
    ))
  }
  </div>
);
