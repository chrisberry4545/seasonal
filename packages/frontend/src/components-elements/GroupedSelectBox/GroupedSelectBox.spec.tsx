import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { GroupedSelectBox } from './GroupedSelectBox';
import { IGroupedSelectOptions } from '@chrisb-dev/seasonal-shared-models';
import { BareButton } from '@chrisb-dev/seasonal-shared-frontend-components';

jest.mock('@chrisb-dev/seasonal-shared-frontend-components', () => ({
  BareButton: () => 'BareButton',
  TextHeadingSmall: () => 'TextHeadingSmall',
  TextMedium: () => 'TextMedium'
}));

describe('<GroupedSelectBox />', () => {
  let wrapper: ShallowWrapper;
  const groups = [{
    groupName: 'g1',
    selectOptions: [{
      name: 'n1',
      value: 'v1'
    }]
  }, {
    groupName: 'g2',
    selectOptions: [{
      name: 'n2',
      value: 'v2'
    }]
  }] as IGroupedSelectOptions[];
  let mockOnSelected: jest.Mock;

  beforeEach(() => {
    mockOnSelected = jest.fn();
    wrapper = shallow(
      <GroupedSelectBox
        onSelected={mockOnSelected}
        groups={groups} />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('can select items in the select box', () => {
    const onClick = wrapper.find(BareButton).first().props().onClick;
    if (onClick) {
      onClick();
    }
    expect(mockOnSelected).toHaveBeenCalledWith('v1');
  });
});
