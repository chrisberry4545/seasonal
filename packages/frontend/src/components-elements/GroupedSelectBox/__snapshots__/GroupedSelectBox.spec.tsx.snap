// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupedSelectBox /> renders correctly 1`] = `
<div>
  <div
    className="c-grouped-select-box__group"
    key="g1"
  >
    <TextHeadingSmall
      className="c-grouped-select-box__group-name"
    >
      g1
    </TextHeadingSmall>
    <BareButton
      className="c-grouped-select-box__option"
      data-e2e="select-option-n1"
      key="v1"
      onClick={[Function]}
    >
      <TextMedium>
        n1
      </TextMedium>
    </BareButton>
  </div>
  <div
    className="c-grouped-select-box__group"
    key="g2"
  >
    <TextHeadingSmall
      className="c-grouped-select-box__group-name"
    >
      g2
    </TextHeadingSmall>
    <BareButton
      className="c-grouped-select-box__option"
      data-e2e="select-option-n2"
      key="v2"
      onClick={[Function]}
    >
      <TextMedium>
        n2
      </TextMedium>
    </BareButton>
  </div>
</div>
`;
